### PR TITLE
fix(ota): version guard firmware updates and sync websocket telemetry

### PR DESCRIPTION
**Pull request by firstof9**
## Description

This PR addresses firmware update verification and state synchronization by:
1. Guarding the `update_firmware` method to only run on firmware versions `4.1.7` or higher (when the firmware introduced `ota_progress` and dynamic `ota` status tracking), raising an `UnsupportedFeature` exception on older versions.
2. Syncing `ota_update`, `ota_progress`, and `ota_state` dynamically when WebSocket updates containing `"ota"` events are received.
3. Ensuring that `ota_update` consistently returns a boolean value.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

* **New Features**
  * Added OTA progress and state properties for real-time monitoring of over-the-air updates
  * Device websocket connections now provide live OTA status updates

* **Improvements**
  * Firmware updates now require minimum device firmware version 4.1.7

### CODE REVIEW

File: openevsehttp/client.py [hunk 1]
```diff
@@ -337,6 +337,15 @@ async def _update_status(self, msgtype, data, error):
...
+
+            if "ota" in keys:
+                ota_val = data["ota"]
+                if ota_val == "started":
+                    self._status["ota_update"] = 1
+                elif ota_val in ("completed", "failed"):
+                    self._status["ota_update"] = 0
+                    self._status.pop("ota_progress", None)
+
             self._status.update(data)
```
[the remark this answers is not in the document]
Comment (firstof9) [reply openevsehttp/client.py:350]: Fixed by popping `ota_progress` from the incoming `data` dict as well to ensure it is not re-added during `_status.update(data)` on completion or failure. Added corresponding unit tests.

**Review by secondof9 — COMMENTED** [review]
## Hermes Agent — Strict Line-by-Line Code Review

**PR:** #591 — fix(ota): version guard firmware updates and sync websocket telemetry
**Verdict:** Approved ✅ (0 issues, 0 suggestions)

---

### ✅ Summary / `openevsehttp/client.py` (lines +337–+347)
- **L+340** — `if "ota" in keys:` — cleanly guards dynamic OTA state extraction without triggering unrelated UPDATE_TRIGGERS.
- **L+341–+342** — `ota_val = data["ota"]` and `if ota_val == "started": self._status["ota_update"] = 1` — correctly maps the `started` event to active update state.
- **L+343–+345** — `elif ota_val in ("completed", "failed"): self._status["ota_update"] = 0; data.pop("ota_progress", None); self._status.pop("ota_progress", None)` — idempotently clears progress when the update ends. The `data.pop`/`self._status.pop` are defensive against stale or replayed events.

**Edge cases / security:** No unstructured JSON exposure; OTA state is internal-only. No secrets or credentials touched. The logic is stateless and thread-safe under asyncio.

---

### ✅ Tests / `tests/test_client.py` (lines +1651–+1677)
- **L+1651–+1654** — `test_update_status_ota()` — cleanly exercises the three OTA lifecycle events (started → completed/failed). The `caplog` fixture is correctly used for logging assertions.
- **L+1661–+1662** — `charger._status = {"ota_update": 0}` — proper isolation between test cases.
- **L+1670–+1673** — `assert charger.ota_progress is None; assert charger.ota_state == "completed"` — correctly validates that progress is cleared on completion, even if the data dict still carries stale `ota_progress`. This is a good edge-case test.

---

### ✅ Command / `openevsehttp/commands.py` (lines +447–+451)
- **L+450–+451** — `if not self._version_check("4.1.7"): _LOGGER.debug(...); raise UnsupportedFeature` — version gating prevents unsupported operations on older firmware. The debug log provides observability without leaking sensitive data.

**Edge cases / security:** No unstructured JSON exposure; firmware update is internal. No secrets or credentials touched. The version check is a pure API contract guard.

---

### ✅ Properties / `openevsehttp/properties.py` (lines +341–+353)
- **L+344** — `return bool(self._status.get("ota_update", False))` — explicit `bool()` avoids Python quirk where `{"ota_update": 1}` evaluates truthy but `True` vs `1` matters in some callers. This is a correctness fix.
- **L+347–+348** — `ota_progress` and `ota_state` — correctly expose the underlying telemetry. The `| None` type hints signal that missing telemetry is a valid state.

**Edge cases / security:** No unstructured JSON exposure; OTA state is internal-only. No secrets or credentials touched.

---

### ✅ Test Setup / `tests/test_commands.py` (lines +1047, +1064, +1083–+1091, +1156, +1167, +1176–+1187)
- **L+1050, +1086, +1178** — `test_charger._config["version"] = "4.1.7"` — consistent mock setup for version-gated tests. The test `test_update_firmware_unsupported` correctly exercises the `UnsupportedFeature` exception path.

**Edge cases / security:** No unstructured JSON exposure; no secrets or credentials touched.

---

### ✅ Test Properties / `tests/test_properties.py` (lines +1040–+1071)
- **L+1043** — Changed expected from `0` to `False` — correctly asserts the boolean nature of `ota_update`.
- **L+1049** — `assert status is expected` — strict `is` comparison validates boolean semantics.
- **L+1052–+1066** — `test_ota_properties()` — cleanly exercises all three OTA properties under the same charger instance. No state leaks between assertions.

**Edge cases / security:** No unstructured JSON exposure; no secrets or credentials touched.

---

### 📌 Summary
All changes are well-structured, fully tested, and aligned with the PR description. No issues or suggestions — clean LGTM.

*Reviewed by Hermes Agent*
